### PR TITLE
Rake task that adds JSON ELM to MAT packages.

### PR DESCRIPTION
**Pull request by hossenlopp**
bonnie:cql:add_json_to_package task takes a MAT package zip and builds a new one in the same directory that has the JSON ELM added. It handles the odd situations of older packages without clause level annotations by replacing the XML ELM with stuff from the translation service.

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1156
- [x] JIRA ticket links to this PR 
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced. N/A
- [x] Tests are included and test edge cases (N/A, This is a utility rake task that would be immensely burdensome to write tests for.)
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered. 
     * See [front-end testing instructions](https://github.com/projecttacoma/bonnie/wiki/Testing#frontend-testing) for getting coverage for front-end tests
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: This is a utility rake task that would be immensely burdensome to write tests for.
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 
- [x] Automated regression test(s) pass N/A, doesn't change calculation code.

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: N/A
- [x] Justification for using JIRA tests: N/A
- [x] JIRA tests have been added to sprint N/A, but this was used to update JIRA tests.

**Reviewer 1:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

**Review by lukoktonos — DISMISSED** [review]
Code looks good.
New .zip file has identical files except for additional json.  
Measure behaves appropriately after uploading it into Bonnie.
Practice deployment used these measures and all selected tests passed.

Approved.

**Review by lukoktonos — DISMISSED** [review]
Changes look good.  Tested with both bash and zsh--working as expected.

File: lib/tasks/cql.rake [hunk 1]
```diff
@@ -333,5 +333,134 @@ namespace :bonnie do
...
+      files = Measures::CqlLoader.get_files_from_zip(File.new(input_package_path), temp_path)
+      raise Exception.new("Package already has ELM JSON!") if files[:ELM_JSON].length > 0
+
+      # translate_cql_to_elm
+      elm_jsons, elm_xmls = CqlElm::CqlToElmHelper.translate_cql_to_elm(files[:CQL])
+
+      # older packages don't have annotations or clause level annotations, if they dont have them wipe out the existing
+      # XML ELM and use the ones from the translation server
+      # start by assuming there are annotations, then set this to false
+      annotations_exist = true
```
Comment (jbradl11) [inline lib/tasks/cql.rake:374]: Should we just always update the XML so we know it matches the JSON? That way we know both artifacts were generated with the same version of the translation service.

Comment (hossenlopp) [reply lib/tasks/cql.rake:374]: The rake task can take an optional parameter to force XML replacement. Should I remove or flip that option and make it default to replace XML?

Comment (jbradl11) [reply lib/tasks/cql.rake:374]: I think that is the correct way to go. But if you have a good reason not to change it, let me know.

Comment (hossenlopp) [reply lib/tasks/cql.rake:374]: Which is the correct way to go? Is it removing the ability to keep the original XML ELM?

I'd argue it should be kept. So we can optionally test the XML ELM because we need to support it anyways.

**Comment by hossenlopp** [comment]
Updated branch and made it so replacing the XML ELM is the default operation.

**Review by lukoktonos — APPROVED** [review]
Latest changes look good.

**Review by jbradl11 — APPROVED** [review]
Code looks good, ran a few times using different packages while passing in different flags. .
